File: go/sieve/sieve.go
package sieve

func Sieve(limit int) []int {
	notPrime := make([]bool, limit-1)
	var primes []int

	for idx := 2; idx <= limit; idx++ {
		if !notPrime[idx-2] {
			primes = append(primes, idx)
			for n := idx * 2; n <= limit; n = n + idx {
				notPrime[n-2] = true
			}
		}
	}

	return primes
}

File: cpp/sieve/sieve.cpp
#include "sieve.h"
#include <vector>

std::vector<int> sieve::primes(size_t num) {
    std::vector<bool> sieve(num + 1, true);

    for (size_t i = 2; i < sieve.size(); i++) {
        if (sieve[i]) {
            for (size_t j = i * 2; j < sieve.size(); j += i) {
                sieve[j] = false;
            }
        }
    }

    std::vector<int> output;
    for (size_t i = 2; i < sieve.size(); i++) {
        if (sieve[i]) {
            output.push_back(i);
        }
    }

    return output;
}
